buildtest/cli/build.py

```diff
@@ -566,16 +566,21 @@ def __init__(
         if timeout:
             if not isinstance(timeout, int):
                 raise BuildTestError(f"{timeout} is not of type int")
-
             if timeout <= 0:
                 raise BuildTestError("Timeout must be greater than 0")
 
         if limit is not None:
             if not isinstance(limit, int):
-                raise BuildTestError(f"{timeout} is not of type int")
+                raise BuildTestError(f"{limit} is not of type int")
             if limit <= 0:
                 raise BuildTestError("Limit must be greater than 0")
 
+        if retry is not None:
+            if not isinstance(retry, int):
+                raise BuildTestError(f"{retry} is not of type int")
+            if retry <= 0:
+                raise BuildTestError("Retry must be greater than 0")
```

Comment:
so do a for loop something like and use the same if condition and just replace the entry with the `field`. Semantically it is the same code just more efficient.

```python
     for field in [timeout, limit, retry]:
        if field is not None:
           ...